src/EDI/Encoder.php

```diff
@@ -31,41 +64,80 @@ public function encode($arr, $wrap = true)
                 $edistring .= "\n";
             }
         }
+        $this->output = $edistring;
         return $edistring;
     }
 
-    public function encodeSegment($row)
-    {
-        $str='';
+    public function encodeSegment($row) {
+        $str = '';
         $t = count($row);
-        for ($i=0; $i<$t; $i++) {
+        for ($i = 0; $i < $t; $i++) {
             $elm = '';
             if (!is_array($row[$i])) {
                 $elm = $this->escapeValue($row[$i]);
             } else {
                 foreach ($row[$i] as &$temp) {
                     $temp = $this->escapeValue($temp);
+                    //var_dump($temp);
                 }
-                $elm = implode(":", $row[$i]);
+                $elm = implode($this->sep_comp, $row[$i]);
             }
             $str .= $elm;
-            if ($i == $t-1) {
+            if ($i == $t - 1) {
                 break;
             }
-            $str .= "+";
+            $str .= $this->sep_data;
         }
-        $str .= "'";
+        $str .= $this->symb_end;
         return $str;
     }
 
-    private function escapeValue($str)
-    {
+//    private function escapeValue($str) {
```

Comment:
just a cancel ommited, i should rewrite this function, it's not clear with those escape chars